README.md

```diff
@@ -1,2 +1,114 @@
+ 2. The `Closable` result from this `activate()` method will deactviate the given `Span`
+    if needed. However, the `Spans` returned by the `GlobalTracer` will also make sure
+    they get deactivated whenever they're finished or closed.
+ 3. A method `GlobalSpanManager.activeSpan()` that will return the _active span_ 
+    from anywhere in the code. This may be useful for libraries that wish to add tags
+    or log information in the active span.
+ 4. It contains a default implementation that uses `ThreadLocal` storage to manage
+    the globally _active span_, activating and deactivating it at the appropriate
+    moments (starting / closing spans etc).
+ 5. `Tracer` implementations or applications that require more customized span management
+    can provide their own subclass of `GlobalSpanManager` and either register it 
+    programmatically through the `GlobalSpanManager.registerInstance()` method 
+    or automatically by the Java SPI ServiceLoader via a 
+    `META-INF/services/io.opentracing.global.GlobalSpanManager` service definition.
+
+## How to use this library
+This section tries to give some basic examples on how to use the Global tracer and
+identify some caveats that need to be taken into account.
+
+### From miscelaneous application / library code
+To add tracing functionality to code that cannot manually propagate `Tracer` or `Span` instances
+(e.g. due to API restrictions or absence of dependency injection techniques)
+simply refer to the global tracer instead; any created `Span` will be made `child-of` the currently _active span_.
+````java
+try (Span span = GlobalTracer.tracer().buildSpan("someOperation").start()) {
+    
+    // ..traced code block here..
+    // calls to GlobalSpanManager.activeSpan() here get the new span reference
+    // the new span is automatically a 'child-of' any previously active span (if applicable).
+
+}
+// calls to GlobalSpanManager.activeSpan() here will receive the active span from before the try block.
+````
+Due to the `try-with-resources` construct, the span will automatically close and restore the _active span_ state to what it was previously.  
+It should be noted that `Tracer` implementations need not be concerned with this functionality;
+it is provided by the `GlobalSpanManager` being invoked from the global tracer.
+
+### From inbound filters at the application boundary (server)
+To create a generic _opentracing_ compatible filter, the hypothetical filter could look similar to
+the following fragment:
+````java
+public void doFilter(someContext, someFilterChain) {
+    // Obtain the inbound context:
+    Format<Carrier> someFormat = ... // usually known by filter
+    Carrier someCarrier = ... // Usually obtained from someContext
+    SpanContext inboundContext = GlobalTracer.tracer().extract(someFormat, someCarrier);
+    
+    // Create a new span as child of the inbound context.
+    try (Span span = GlobalTracer.tracer().buildSpan("inboundOperation")
```

Comment:
@yurishkuro Correct, in that case the example would be split in two effectively, finishing it in the response interceptor.
The `GlobalSpanManager.clearAllSpans()` is actually specifically intended for this situation as a safety net to make sure threads are returned to threadpools without dangling spans in there (because the try-with-resources guarantee is not there in the split case).